crates/hotshot/src/traits/election/randomized_committee_members.rs

```diff
@@ -0,0 +1,250 @@
+    fn stake(
+        &self,
+        pub_key: &<TYPES as NodeType>::SignatureKey,
+        epoch: <TYPES as NodeType>::Epoch,
+    ) -> Option<<TYPES::SignatureKey as SignatureKey>::StakeTableEntry> {
+        let filter = self.make_quorum_filter(epoch);
+        let actual_members: BTreeSet<_> = self
+            .stake_table
+            .iter()
+            .enumerate()
+            .filter(|(idx, _)| filter.contains(idx))
+            .map(|(_, v)| TYPES::SignatureKey::public_key(v))
+            .collect();
+
+        if actual_members.contains(pub_key) {
+            // Only return the stake if it is above zero
+            self.indexed_stake_table.get(pub_key).cloned()
+        } else {
+            // Skip members which aren't included based on the quorum filter
+            None
+        }
+    }
+
+    /// Check if a node has stake in the committee
+    fn has_stake(
+        &self,
+        pub_key: &<TYPES as NodeType>::SignatureKey,
+        epoch: <TYPES as NodeType>::Epoch,
+    ) -> bool {
+        let filter = self.make_quorum_filter(epoch);
+        let actual_members: BTreeSet<_> = self
+            .stake_table
+            .iter()
+            .enumerate()
+            .filter(|(idx, _)| filter.contains(idx))
+            .map(|(_, v)| TYPES::SignatureKey::public_key(v))
+            .collect();
+
+        if actual_members.contains(pub_key) {
+            self.indexed_stake_table
+                .get(pub_key)
+                .is_some_and(|x| x.stake() > U256::zero())
+        } else {
+            // Skip members which aren't included based on the quorum filter
+            false
+        }
+    }
+
+    /// Get the network topic for the committee
+    fn committee_topic(&self) -> Topic {
+        self.committee_topic.clone()
+    }
+
+    /// Index the vector of public keys with the current view number
+    fn lookup_leader(
+        &self,
+        view_number: TYPES::View,
+        epoch: <TYPES as NodeType>::Epoch,
+    ) -> Result<TYPES::SignatureKey> {
+        let filter = self.make_quorum_filter(epoch);
+        let leader_vec: Vec<_> = self
+            .stake_table
+            .iter()
+            .enumerate()
+            .filter(|(idx, _)| filter.contains(idx))
+            .map(|(_, v)| v.clone())
+            .collect();
+
+        let mut rng: StdRng = rand::SeedableRng::seed_from_u64(*view_number);
+
+        let randomized_view_number: u64 = rng.gen_range(0..=u64::MAX);
+        #[allow(clippy::cast_possible_truncation)]
+        let index = randomized_view_number as usize % leader_vec.len();
+
+        let res = leader_vec[index].clone();
+
+        Ok(TYPES::SignatureKey::public_key(&res))
+    }
+
+    /// Get the total number of nodes in the committee
+    fn total_nodes(&self, epoch: <TYPES as NodeType>::Epoch) -> usize {
+        self.make_quorum_filter(epoch).len()
+    }
+
+    /// Get the voting success threshold for the committee
+    fn success_threshold(&self) -> NonZeroU64 {
```

Comment:
I think these functions need to take `epoch` as an argument, because the success/failure thresholds (as an absolute rather than a percentage) depends on the membership size in that epoch

rather than `stake_table` I think we need to call `stake_table()`

Reply:
oh actually, nevermind -- I see what you did!

yes I think this should just be a separate PR; it's correct as-is the way you wrote it